doc/headset.md

```diff
@@ -0,0 +1,430 @@
+            break;
+        case 'deviceConnectionStatusChanged':
+            this.set('headsetState', {
+                isConnected: value.payload.isConnected,
+                isConnecting: value.payload.isConnecting
+            });
+            anyNecessaryFunctions(value.payload.isConnected, value.payload.isConnecting)
+            break;
+        default:
+        // console.log('some other event');
+}
+
+```
+
+### Methods
+
+#### `updateAudioInputDevice()`
+Function that will update what the active mic is in the headset library. This passes the newly selected mic label into the headset library where it will check to see which (if any) of the currently supported implementations will be a best fit for the new device This function should be called any time a new audio device is selected.
+
+This function is currently called from `client.ts` if a new audio device is selected through the function `sdk.updateDefaultDevices()`.
+
+This function will call through `sdk.media.findCachedDeviceByIdAndKind(newMicId)` which takes in the newly selected device's ID and will return the device's complete information including the necessary device label
+
+After receiving the complete device information, it will send the device's label to the headset library through `this.headsetLibrary.activeMicChange(completeDeviceInfo.label.toLowerCase())`
+
+Declaration:
+``` ts
+    updateAudioInputDevice (newMicId: string) : void;
+```
+
+Params:
+* `newMicId: string` - The ID of the newly selected device sent from the `client.ts`
+
+Returns: void
+
+#### `getCurrentSelectedImplementation()`
+Function that fetches the currently selected implementation from the headset library. This will help with checking proper flags for various functionality such as `this.getCurrentSelectedImplementation().isConnected`
+
+Declaration:
+``` ts
+    getCurrentSelectedImplementation () : VendorImplementation;
+```
+
+Params: none
+
+Returns: an instance of a VendorImplementation from the headset library
+
+#### `showRetry()`
+Function that determines if a "show retry" button should be shown on screen to allow the user to attempt to reconnect to their selected vendor and the corresponding implementation. The show retry button should show up if:
+    1.) The selected implementation has disabled retry capabilities
+    2.) The selected device is not supported (`!selectedImplementation`)
+    3.) The selected implementation is not connected (`!selectedImplementation.isConnected`)
+    4.) The selected implementation is not connecting (`!selectedImplementation.isConnecting`)
+
+Declaration:
+``` ts
+    showRetry (): boolean;
+```
+
+Params: none
+
+Returns: a boolean value determining if a the show retry button is required
+
+#### `retryConnection()`
+Function to retry connection to the selected implementation. It takes in the micLabel of the device in question and calls the `connect(label)` function of the corresponding implementation.
+    NOTE: This may be reworked
```

Comment:
At this moment there are technically two implementations of this function.  One in the SDK that hits the selected implementation's `connect()` function on its own and another one in the headset library that does the same thing but also provides the opportunity to print out an error if need be.  I left this note because I wasn't too sure what one was the better approach. On one hand we're supplying them a function with everything they need out of the box.  On the other hand, what if they wanted a specific implementation for their `retryConnection`

Reply:
Does this "may be reworked" still apply with the changes you just pushed to the headset lib?